### PR TITLE
feat(arc-575): get playable urls for flowplayer

### PR DESCRIPTION
**Pull request by BavoVdg**
Playable url opgehaald voor flowplayer

### CODE REVIEW

File: src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx [hunk 1]
```diff
@@ -198,14 +211,18 @@ const ObjectDetailPage: NextPage = () => {
...
+								<FlowPlayer
+									className="p-object-detail__flowplayer"
+									src={playableUrl}
+									poster="https://via.placeholder.com/1920x1080"
+									title="Elephants dream"
+									pause={pauseMedia}
+									onPlay={() => setPauseMedia(false)}
+								/>
+							) : (
+								<></>
```
Comment (andrych) [inline src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx:224]: Als je niets wil renderen kan je gwn `null` returnen

File: src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx [hunk 2]
```diff
@@ -45,6 +46,7 @@ const ObjectDetailPage: NextPage = () => {
 	const [activeBlade, setActiveBlade] = useState<MediaActions | null>(null);
 	const [mediaType, setMediaType] = useState<MediaTypes>(null);
 	const [pauseMedia, setPauseMedia] = useState(true);
+	const [mediaUrl, setMediaUrl] = useState<string | undefined>(undefined);
```
Comment (andrych) [inline src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx:49]: Je hoeft hier niet `undefined` mee te geven `useState()` is hetzelfde als `useState(undefined)`

Comment (andrych) [inline src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx:49]: Je hoeft hier niet `undefined` mee te geven `useState()` is hetzelfde als `useState(undefined)`

**Review by andrych — APPROVED** [review]

File: src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx [hunk 3]
```diff
@@ -63,10 +65,21 @@ const ObjectDetailPage: NextPage = () => {
 		router.query.objectId as string
 	);
 
-	// Set default view
+	const { data: playableUrl, isLoading: isLoadingPlayableUrl } = useGetMediaTicketInfo(
+		mediaUrl ?? '',
+		!!mediaUrl
+	);
```
Comment (bertyhell) [inline src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx:71]: you can also handle that in the media service in the getPlayableUrl function. That way if any other code calls the mediaService.getPlayableUrl directly that will also have the benefit of this check:

```
	public static async getPlayableUrl(
		referenceId: string | null
	): Promise<string | null> {
	        if (!referenceId) {
		       return null;
	        }
		return await ApiService.getApi()
			.get(
				stringifyUrl({
					url: `${MEDIA_SERVICE_BASE_URL}/${MEDIA_SERVICE_TICKET_URL}`,
					query: {
						id: referenceId,
					},
				})
			)
			.text();
	}
```

File: src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx [hunk 4]
```diff
@@ -63,10 +65,21 @@ const ObjectDetailPage: NextPage = () => {
 		router.query.objectId as string
 	);
 
-	// Set default view
+	const { data: playableUrl, isLoading: isLoadingPlayableUrl } = useGetMediaTicketInfo(
```
Comment (bertyhell) [inline src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx:68]: maybe also handle isError, so we can show some message like:
```
	Het ophalen van de video afspeel token is mislukt.
```

File: src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx [hunk 5]
```diff
@@ -198,14 +211,16 @@ const ObjectDetailPage: NextPage = () => {
...
-								title="Elephants dream"
-								pause={pauseMedia}
-								onPlay={() => setPauseMedia(false)}
-							/>
+							!isLoadingPlayableUrl && playableUrl ? (
+								<FlowPlayer
+									className="p-object-detail__flowplayer"
+									src={playableUrl}
+									poster="https://via.placeholder.com/1920x1080"
+									title="Elephants dream"
```
Comment (bertyhell) [inline src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx:219]: Title isn't correct anymore. Pass nothing or maybe the object name?

**Review by bertyhell — APPROVED** [review]

File: src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx (hunk 3 above)
Comment (ian-emsens-shd) [reply src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx:74]: `!!mediaUrl` will prevent the mediaservice from being called until the actual url is available to the view, IMO the service should dictate its requirements, not fire empty promises while its waiting for actual input

**Review by ian-emsens-shd — APPROVED** [review]

File: src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx (hunk 5 above)
Comment (BavoVdg) [reply src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx:215]: Fixed in future pr

**Pull request merged**